pkg/apis/serving/v1beta1/component.go

```diff
@@ -88,11 +90,17 @@ type ComponentExtensionSpec struct {
 	// (https://knative.dev/docs/serving/autoscaling/autoscaling-targets/).
 	// +optional
 	ScaleTarget *int `json:"scaleTarget,omitempty"`
+	// AutoScaling to be used for autoscaling spec. Could be used for Keda autoscaling.
+	// +optional
+	AutoScaling []AutoScalingSpec `json:"autoScaling,omitempty"`
```

Comment:
AutoScalingSpec should not be an array, it contains metrics array and autoscaling behavior which we can add later.